supabase/functions/process-audio/index.ts

```diff
@@ -35,85 +35,141 @@ const processAudio = async (req: Request) => {
+    ) {
+      return new Response(JSON.stringify({ message: "No transcript found." }), {
+        headers: { ...corsHeaders, "Content-Type": "application/json" },
+        status: 200,
+      });
+    }
 
-        const { data, error } = await supabase
-            .from('records')
-            .insert({ raw_text: transcript, embeddings: embeddings });
+    console.log("Transcript:", transcript);
 
-        if (error) {
-            console.error('Error inserting record:', error);
-        }
-    } catch (error) {
-        console.error('Transcription error:', error);
-        return new Response(JSON.stringify({ error: error.message }), {
-            headers: { ...corsHeaders, 'Content-Type': 'application/json' },
-            status: 500,
-        });
+    const { data: unprocessedRecords, error: fetchError } = await supabase
+    .from('records')
+    .select('raw_text, id')
+    .eq('processed', false)
+    .order('created_at', { ascending: true })
+    .limit(4);
+
+    if (fetchError) {
+      console.error("Error fetching records:", fetchError);
+      return new Response(JSON.stringify({ message: "Error fetching records." }), {
+        headers: { ...corsHeaders, "Content-Type": "application/json" },
+        status: 500,
+      });
     }
+  
+    const insertResponse = await supabase
+    .from("records")
+    .insert([{ raw_text: transcript, processed: false }])
+    .select();
 
-    return new Response(
-        JSON.stringify({
-            message: 'Audio transcribed successfully.',
-            transcript,
-        }),
-        {
-            headers: { ...corsHeaders, 'Content-Type': 'application/json' },
-            status: 200,
-        }
-    );
+    if (insertResponse.error) {
+      console.error("Error inserting record:", insertResponse.error);
+      return new Response(JSON.stringify({ message: "Error inserting record." }), {
+        headers: { ...corsHeaders, "Content-Type": "application/json" },
+        status: 500,
+      });
+    }
+
+    const currentTranscript = {
+      raw_text: transcript,
+      id: insertResponse.data[0].id,
+    };
+
+    const allRecordsToProcess = [...unprocessedRecords, currentTranscript];
+    if (allRecordsToProcess.length === 5) {
+      console.log("Processing new transcript with 4 unprocessed records...");
+      let concatenatedTranscripts = unprocessedRecords
+        .map((record: any) => record.raw_text)
+        .join(" ");
+
+      const response = await openaiClient.chat.completions.create({
+        model: 'gpt-4-0125-preview',
+        messages: [
+            {
+                role: 'system',
+                content: `
+        These transcripts contain information about your user. 
+        Your task is to organize the information in a way that makes sense to you.
+        Your response must be in json format with the three following keys: "summary", "topics".
+        `,
+            },
+            {
+                role: 'user',
```

Comment:
There is no need for system-user-assistant messages here.

Just make the system hold the full requirements, and let the assistant reply

Reply:
Ya there is, according to the docs. If you use json mode you need to add a system message explaining what you want

Reply:
Yeah that is fine, what I meant is that there is no need for the `system-user-assistant` , we only need `system-assistant` (i.e. there is no need for "user" in this flow)